src/Adapter/Product/GeneralConfiguration.php

```diff
@@ -68,7 +69,9 @@ public function updateConfiguration(array $config)
         $errors = [];
 
         if ($this->validateConfiguration($config)) {
-            $this->configuration->set('PS_CATALOG_MODE', (int) $config['catalog_mode']);
+            $catalogMode = (int) $config['catalog_mode'];
+            $this->configuration->set('PS_CATALOG_MODE', $catalogMode);
+            $this->configuration->set('PS_CATALOG_MODE_WITH_PRICES', $catalogMode ? (int) $config['catalog_mode_with_prices'] : 0);
```

Comment:
Isn't it weird for the merchant to have another value to be set? What would be the consequence to only rely on `$config['catalog_mode_with_prices']`?

Reply:
the `PS_CATALOG_MODE_WITH_PRICES` is only taken into account when `PS_CATALOG_MODE` is on Otherwise it has no use (that's why the toggle button is hidden in the BO when catalog mode is off)